backend/src/main/java/com/google/sps/datastore/ParticipantDatastore.java

```diff
@@ -0,0 +1,150 @@
+    Key participantKey = KeyFactory.createKey(KIND_PARTICIPANT, username);
+    try {
+      return datastore.get(participantKey);
+    } catch (EntityNotFoundException e) {
+      // TODO: what error to send
+      return null;
+    }
+  }
+
+  /** Update Participant with new matchId, null out availability to show currently matched */
+  public void updateNewMatch(String username, long matchId) {
+    Entity participantEntity = getEntityFromUsername(username);
+
+    participantEntity.setProperty(PROPERTY_CURRENTMATCHID, matchId);
+
+    // Null out availibility fields if not already
+    participantEntity.setProperty(PROPERTY_STARTTIMEAVAILABLE, null);
+    participantEntity.setProperty(PROPERTY_ENDTIMEAVAILABLE, null);
+    participantEntity.setProperty(PROPERTY_DURATION, 0);
+
+    // Overwrite existing entity in datastore
+    datastore.put(participantEntity);
+  }
+
+  /** Return Participant from username */
+  @Nullable
+  public Participant getParticipantFromUsername(String username) {
+    return getParticipantFromEntity(getEntityFromUsername(username));
+  }
+
+  /** Return participant object from datastore participant entity */
+  private Participant getParticipantFromEntity(Entity entity) {
+    // Get entity properties
+    String username = (String) entity.getProperty(PROPERTY_USERNAME);
+    ZonedDateTime startTimeAvailable =
+        ZonedDateTime.parse((String) entity.getProperty(PROPERTY_STARTTIMEAVAILABLE), formatter);
+    ZonedDateTime endTimeAvailable =
+        ZonedDateTime.parse((String) entity.getProperty(PROPERTY_ENDTIMEAVAILABLE), formatter);
+    int duration = ((Long) entity.getProperty(PROPERTY_DURATION)).intValue();
+    long currentMatchId = (long) entity.getProperty(PROPERTY_CURRENTMATCHID);
+    long timestamp = (long) entity.getProperty(PROPERTY_TIMESTAMP);
+
+    // Create and return new Participant
+    return new Participant(
+        username, startTimeAvailable, endTimeAvailable, duration, currentMatchId, timestamp);
+  }
+
+  /** Return list of all unmatched participants */
+  public List<Participant> getUnmatchedParticipants() {
+    Query query = new Query(KIND_PARTICIPANT).addSort(PROPERTY_DURATION, SortDirection.DESCENDING);
+
+    // Create filter to get all currently unmatched participants
+    Filter unmatched = new FilterPredicate(PROPERTY_DURATION, FilterOperator.GREATER_THAN, 0);
+    query.setFilter(unmatched);
+
+    PreparedQuery results = datastore.prepare(query);
+
+    // Convert entities to list of participants
+    List<Participant> participants = new ArrayList<Participant>();
+    for (Entity entity : results.asIterable()) {
+      participants.add(getParticipantFromEntity(entity));
+    }
+    return participants;
+  }
+
+  /** Remove Participant from datastore */
+  public void removeParticipant(String username) {
+    Key participantKey = KeyFactory.createKey(KIND_PARTICIPANT, username);
+    datastore.delete(participantKey);
+  }
```

Comment:
Since we are moving away from deleting participants and rather nulling out time/availability fields, will this method still need to be included?